### ISSUE TITLE
Android build fail because required parameters are not passed

### ISSUE THREAD

**Issue by davidmfinol** [game-ci/unity-builder#334]
**Bug description**

Builds from main branch fail for Android because the required Android parameters are no longer passed

**How to reproduce**

- Run an Android build using `game-ci/unity-builder@main`
- Build errors

**Expected behavior**

Build should succeed

**Additional details**

Example failing build: https://github.com/finol-digital/Card-Game-Simulator/runs/5065670391?check_suite_focus=true#step:6:377

In particular, it seems that `ANDROID_KEYSTORE_BASE64` and `ANDROID_SDK_MANAGER_PARAMETERS` are not going through.

EDIT: The break is from within the past week. Not sure if it was cloud runner v2 or the initial mac build support that has the regression: https://github.com/game-ci/unity-builder/compare/v2.0.0-alpha-14...main
The test workflows all passed for both of those, as we don't have a scenario for Android builds that require a custom keystore (maybe I should create a separate issue for adding this test scenario?).

**Comment by webbertakken** [comment]
At this point we should probably start thinking about extending our tests and possible restructuring some of the code to become more managable.

The index file has become a bit of a mess. We should probably use a strategy pattern for the different platforms and have each builder go through a sequence of steps that will guarantee some level of consistency between the different builder implementations.

If we have that it would be easier for contributors to plug in code to their builder implementation and add tests.

**Comment by webbertakken** [comment]
I checked AndroidKeystoreBase64 which seems to be passed through all files correctly.

In your workflow it looks like ˋnameˋ from the input class is being used https://github.com/finol-digital/Card-Game-Simulator/runs/5065670391?check_suite_focus=true#step:6:91

It says ˋname Inputˋ.

**Comment by davidmfinol** [comment]
You can see that `AndroidKeystoreBase64` is indeed received as input: [Log 1](https://github.com/finol-digital/Card-Game-Simulator/runs/5065670391?check_suite_focus=true#step:6:7)

Then, during [SetupCli](https://github.com/game-ci/unity-builder/blob/main/src/model/cli/cli.ts#L42), it is again showing OK: [Log 2](https://github.com/finol-digital/Card-Game-Simulator/runs/5065670391?check_suite_focus=true#step:6:7)

But in the [docker run command](https://github.com/game-ci/unity-builder/blob/main/src/model/docker.ts#L35), it is showing as empty: [Log 3](https://github.com/finol-digital/Card-Game-Simulator/runs/5065670391?check_suite_focus=true#step:6:249)

And now I see that the gap is actually on this line here: https://github.com/game-ci/unity-builder/blob/a6ebcb5b7518eb5fe9642d6b63e06ef2b41e9e4c/src/model/image-environment-factory.ts#L17-L20

@frostebite 
Why are multi-line values are no longer passed through to docker on line 18 of `image-environment-factory.ts`? `ANDROID_KEYSTORE_BASE64` needs to be multi-line, so can an exception be made here for `ANDROID_KEYSTORE_BASE64`?

**Comment by frostebite** [comment]
Not sure from reading the info attached to this ticket. Will take a deeper look.

**Comment by frostebite** [comment]
Doesn't appear to have been altered significantly at all in cloud runner V2.

Will investigate more.

**Comment by davidmfinol** [comment]
>The flow of these parameters doesn't appear to have been altered significantly at all in cloud runner V2

Right, I believe it is just lines 17-20 of `image-environment-factory.ts` (which was newly created in cloud runner V2) that is causing this error. 
Removing those 4 lines should fix the error, but I don't know if they were added to fix some other issue in cloud runner?

**Comment by frostebite** [comment]
Thank you for the additional clarity. I will look further tomorrow and get
back to you.

**Comment by frostebite** [comment]
Will explore the best way to prevent this with a test case too. :)

Sorry for any disruption.

**Comment by frostebite** [comment]
If this is blocking a release it can be corrected to unblock the android
parameters. And I'm happy to fix for cloud runner separately.

I haven't spent enough time digging into a resolution yet. Will be looking
at it more in the next few days.

**Comment by davidmfinol** [comment]
I went ahead and opened a PR with the fix: https://github.com/game-ci/unity-builder/pull/335
Tested it here: https://github.com/finol-digital/Card-Game-Simulator/runs/5177156811?check_suite_focus=true
@frostebite feel free to merge if you think it's ok

**Comment by davidmfinol** [comment]
I went ahead and merged the PR. If it breaks something in cloud runner, we can fix in another issue/PR.
